docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+.. _path-resolution:
+
+***************
+Path Resolution
+***************
+
+In order to be able to support reproducible builds on all platforms, the Solidity compiler has to
+abstract away the details of the filesystem where source files are stored.
+Paths used in imports must work the same way everywhere while the command-line interface must be
+able to work with platform-specific paths to provide good user experience.
+This section aims to explain in detail how Solidity reconciles these requirements.
+
+.. index:: ! virtual filesystem, ! source unit name
+.. _virtual-filesystem:
+
+Virtual Filesystem
+==================
+
+The paths used in :ref:`imports <import>` in a general case do not have to be filesystem paths.
+The compiler maintains an internal database (virtual filesystem) where each compiled source unit is
+assigned a unique *source unit name* which is an opaque and unstructured identifier.
+
+Source files can be placed in the virtual filesystem directly, using :ref:`Standard JSON interface
+<compiler-api>` and in this case source unit names can be anything.
+In most cases, however, the project files reside in the local filesystem and it is desirable for
+the compiler to be able to locate and load them automatically.
+To support this, the compiler passes all source unit names not found in the virtual filesystem to
+the file loader.
+In case of the command-line compiler the file loader attempts to interpret them as local paths.
```

Comment:
What is a local path? A path relative to the current working directory, or a path in the host filesystem?

Reply:
The latter.

I think the text feels very repetitive and for some concepts I, unfortunately, have very few synonyms I can use without making things ambiguous. But I can go back to filesystem paths here.